src/ImageSharp/Formats/Tiff/PhotometricInterpretation/BlackIsZero4TiffColor{TPixel}.cs

```diff
@@ -24,6 +24,7 @@ public override void Decode(ReadOnlySpan<byte> data, Buffer2D<TPixel> pixels, in
             var l8 = default(L8);
             for (int y = top; y < top + height; y++)
             {
+                Span<TPixel> pixelRowSpan = pixels.DangerousGetRowSpan(y);
```

Comment:
If I understand correctly you suggest:
```
Span<TPixel> pixelRowSpan = pixels.DangerousGetRowSpan(y).Slice(left, left + width - 1);
for (int x = 0; x < pixelRowSpan.Length;)
``` 

That will cause an out of bounds access, because x is incremented twice in the loop.

Also we need the complete row when its odd, see the last condition `if (isOddWidth) {`  ...

Reply:
Hi @brianpopow -
Any ideas when these perf enhancements will land?  I could really use them.

Reply:
@TonyValenti I am sorry, but I cannot give a estimate when this will be merged. This will be merged, when there was a proper review by a member of the ImageSharp team. We are all doing this in our free time / spare time, so it can take a while. You have to be patience.

Reply:
>That will cause an out of bounds access, because x is incremented twice in the loop.

@brianpopow  Slicing to `left + width`  and using 1 less for the max will solve that no? I may still be missing something.

```c#
Span<TPixel> pixelRowSpan = pixels.DangerousGetRowSpan(y).Slice(left, left + width);
for (int x = 0; x < pixelRowSpan.Length - 1;)
```

Reply:
@JimBobSquarePants: As far as I understand this, the bounds checks will only be remove by the Jit, if `pixelRowSpan.Length` and not when `pixelRowSpan.Length - 1` is used as the upper bounds of the loop. I tried to replicate this in [SharpLab](https://sharplab.io/#v2:EYLgHgbALANALiAhgZwLYB8ACAmAjAWAChMBmAAhzIGEyBvIsxi8gSwDs4yBlAV1QAp2cANoBdMgBNEcRAEoGTeoSZMhZZHzIBeMgAYA3ApUAzAPYAnMoI5kW2vfttkAPJOmIAdABkApmwDmcAAWjiwA1GGy6pphOlIywiyihsoqFADs0agpKgC+RmQFpLY2vKjY1iLi8XIFSmklnBqo9gYFjGaWlU46Bk6uNd5+gUEAtLihEVHNZLFuCUk5aZiZzUuM+YS5QA==). I am not so good in reading Jit Asm, but the second method has more instructions.

Reply:
You're not actually comparing the right thing there. 

If you look at the Asm for .NET 6 x64 and you'll see that the only difference is the `dec` instruction which subtracts one from the destination operand which is fine.

```asm
L0008: dec r8d
```

The 3rd method shows what you are doing now and you can see the compare and conditional jump instructions

```asm
L0014: cmp ecx, r8d
L0017: jae short L002e
```

Then a whole bunch of additional instructions

```asm
L0027: jl short L0010
L0029: add rsp, 0x28
L002d: ret
L002e: call 0x00007ffdfa77ec40
L0033: int3
```

https://sharplab.io/#gist:ba6c722dc587c2773ad729530c591045

I'm actually on the lookout for a good guide for assembly instruction as I'm pretty rubbish at reading it myself. Here's the best I've found so far.
https://www.aldeid.com/wiki/Category:Architecture/x86-assembly

Reply:
@JimBobSquarePants there are failing tests when I change it to slicing and I cant figure out what's wrong. I think that the 4- Bit Photometric Interpretation is a niche use case and I would like to keep as it is now. I think the potential gain is not worth the trouble and we may introduce issues with the change.

Reply:
No worries, something to keep in mind for the future anyway. The speed up you’ve produced so far is ace!